presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -85,6 +86,14 @@ default boolean schemaExists(ConnectorSession session, String schemaName)
      */
     ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName);
 
+    /**
+     * Returns an error for connectors which do not support table version AS OF expression.
+     */
+    default ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName, Optional<Type> tableVersionType, Optional<Block> tableVersionExpression)
```

Comment:
would it ever be the case that `tableVersionType` is present without `tableVersionExpression`? AFAICT, there isn't a case where one would be present without the other. 

Unless there is a case, I think it would be better to combine `tableVersionType` and `tableVersionExpression` into a single argument to prevent incorrectly calling this method

Reply:
Fixed